### ISSUE TITLE
[Bug] On behalf of supplier doesn't pass tenant id to silent supplier

### ISSUE THREAD

**Issue by q-benwillis** [AzureAD/microsoft-authentication-library-for-java#881]
### Library version used

1.17.2

### Java version

17.0.11

### Scenario

ConfidentialClient - web api (AcquireTokenOnBehalfOf)

### Is this a new or an existing app?

This is a new app or experiment

### Issue description and reproduction steps

It seems that when the OBO supplier attempts to acquire the token silently from the cache it doesn't pass the Tenant ID in the SilentParameters which means that the token cannot be found in the cache.

I'm sending OnBehalfOfParameters with the Tenant ID:
```
OnBehalfOfParameters.builder(scopes, userAssertion).tenant(tenantId).build
```

This is all I'm seeing in the logs:
```
2024-11-22T11:56:56.127Z DEBUG 24 --- [app] [onPool-worker-3] c.m.a.m.AcquireTokenByOnBehalfOfSupplier : SkipCache set to false. Attempting cache lookup
2024-11-22T11:56:56.128Z DEBUG 24 --- [app] [onPool-worker-3] c.m.a.m.AcquireTokenByOnBehalfOfSupplier : Cache lookup failed: Token not found in the cache
```

However if I dig a little deeper I can see that the cache miss is because the cached token realm doesn't match the authority tenant [here](https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/081341c8d0afe8ca5d024f60047d45c5f440a5c7/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/TokenCache.java#L501). If I follow this back I can see that the authority tenant can come from the request parameters [here](https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/081341c8d0afe8ca5d024f60047d45c5f440a5c7/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AuthenticationResultSupplier.java#L32). However it looks like this isn't happening because the `OnBehalfOfSupplier` doesn't add the tenant to the `SilentParameters` [here](https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/081341c8d0afe8ca5d024f60047d45c5f440a5c7/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenByOnBehalfOfSupplier.java#L28).

### Relevant code snippets

```csharp
val userAssertion = new UserAssertion(???)
val scopes = Set(???)
val tenantId = ???

val parameters = OnBehalfOfParameters.builder(scopes, userAssertion).tenant(tenantId).build

// Call confidential client with below parameters and request should succeed and write OBO token to cache
client.acquireToken(parameters)

// Call confidential client with again with same parameters and request does not find previous token from cache
client.acquireToken(parameters)
```

### Expected behavior

The second request should be able to retrieve the token from the cache

### Identity provider

Microsoft Entra ID (Work and School accounts and Personal Microsoft accounts)

**Comment by Avery-Dunn** [comment]
Hello @q-benwillis, thanks for bringing this to our attention. The root cause was exactly what you described: the tenant wasn't getting added to our internal silent call, and I've just created a PR that should fix it: https://github.com/AzureAD/microsoft-authentication-library-for-java/pull/886

I'll update this thread once it's released, which should be by the end of this week.

**Comment by Avery-Dunn** [comment]
The fix for this was released as part of [1.18.0](https://mvnrepository.com/artifact/com.microsoft.azure/msal4j/1.18.0), thanks again for letting us know about this!
